src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -20,73 +20,51 @@
+import net.minecraft.server.packs.resources.PreparableReloadListener;
+import net.minecraft.server.players.PlayerList;
+import net.minecraft.sounds.SoundEvent;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.Container;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.*;
+import net.minecraft.world.entity.animal.Animal;
 import net.minecraft.world.entity.item.ItemEntity;
 import net.minecraft.world.entity.monster.Zombie;
-import net.minecraft.world.entity.animal.Animal;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.entity.player.Player.BedSleepingProblem;
-import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.entity.projectile.Projectile;
+import net.minecraft.world.entity.projectile.ThrownEnderpearl;
 import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
 import net.minecraft.world.item.context.UseOnContext;
 import net.minecraft.world.item.crafting.RecipeType;
+import net.minecraft.world.level.*;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.portal.PortalShape;
+import net.minecraft.world.level.storage.PlayerDataStorage;
+import net.minecraft.world.level.storage.ServerLevelData;
 import net.minecraft.world.level.storage.loot.LootTable;
 import net.minecraft.world.level.storage.loot.LootTables;
-import net.minecraft.server.packs.resources.PreparableReloadListener;
-import net.minecraft.server.ServerResources;
-import net.minecraft.world.level.BaseSpawner;
-import net.minecraft.world.InteractionResultHolder;
-import net.minecraft.world.InteractionResult;
-import net.minecraft.core.Direction;
-import net.minecraft.world.InteractionHand;
-import net.minecraft.core.NonNullList;
-import net.minecraft.resources.ResourceKey;
-import net.minecraft.resources.ResourceLocation;
-import net.minecraft.sounds.SoundSource;
-import net.minecraft.sounds.SoundEvent;
-import net.minecraft.core.BlockPos;
-import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.phys.HitResult;
-import net.minecraft.network.chat.ChatType;
-import net.minecraft.network.chat.Component;
-import net.minecraft.world.level.Explosion;
-import net.minecraft.world.level.GameRules;
-import net.minecraft.world.level.LevelAccessor;
-import net.minecraft.world.level.LevelReader;
-import net.minecraft.world.level.Level;
-import net.minecraft.server.level.ServerLevel;
-import net.minecraft.world.level.biome.MobSpawnSettings;
-import net.minecraft.world.level.storage.ServerLevelData;
-import net.minecraft.world.level.storage.PlayerDataStorage;
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
 import net.minecraftforge.client.MinecraftForgeClient;
```

Comment:
Please avoid reorganizing imports or using wildcard imports.

Reply:
The wildcard import already existed and is used across multiple files in the forge project (https://i.gyazo.com/26808eefd63ff06786a7ae21d28adc42.png)

Reply:
The fact that it's used elsewhere (which is a mistake) does not change the fact that we require PRs to not *add* them.

Reply:
The fact that it's used elsewhere (which is a mistake) does not change the fact that we require PRs to not *add* them.

Reply:
Well given it's unused in an uncountably infinite number of places across the project already it feels like an arbitrary and futile thing to require. 

In addition to this, if you read what I wrote, my point was that I did NOT add this and as such it's not a worthwhile comment to make.

Reply:
Regardless of the use (or lack thereof) of wildcard imports in other places within the project, the official code style used by Forge enforced for PRs mandates the lack of use of the wildcard import. The existing wildcard import _is_ a mistake (one which we will correct at the next opportunity), and should not be taken as a greenlight to add further wildcard imports.

Furthermore, the more pressing change here is the reorganization of the imports (usually caused by performing the IDE's optimize/organize imports functionality), causing the mess of import changes when reviewing files.

Reply:
Should be fixed with latest commit, just commenting to ensure